state/azure/cosmosdb/cosmosdb_query.go

```diff
@@ -147,9 +147,10 @@ func (q *Query) setNextParameter(val string) string {
 func (q *Query) execute(client *azcosmos.ContainerClient) ([]state.QueryItem, string, error) {
 	opts := &azcosmos.QueryOptions{}
 
+	resultLimit := q.limit
 	opts.QueryParameters = append(opts.QueryParameters, q.query.parameters...)
 	if q.limit != 0 {
-		opts.PageSizeHint = int32(q.limit)
+		opts.PageSizeHint = int32(resultLimit)
```

Comment:
Let's not set the Page Size at all - let's rely on the default. We iterate through all pages anyway to fetch all results.